config.gemspec

```diff
@@ -28,6 +28,7 @@ Gem::Specification.new do |s|
   s.required_ruby_version = '>= 2.6.0'
 
   s.add_dependency 'deep_merge', '~> 1.2', '>= 1.2.1'
+  s.add_dependency 'ostruct' if RUBY_VERSION >= '3.3.5'
```

Comment:
I don't think you can do conditionals like this in a gemspec, because when you gem package it evaluates this at package time.  That is, if it's packaged on < 3.3.5 it won't exist, and if it's packaged on 3.3.5+ it will exist, which means it's non-deterministic.

Reply:
@Fryguy  Thank you for a comment.
I removed condition.